ARCs/arc-0018.md

```diff
@@ -0,0 +1,348 @@
+            "desc":"Create an nft with appropriate parameters, return its id",
+            "args":[],
+            "returns":{"name":"asset_id", "type":"uint64"}
+        },
+        {
+            "name":"offer",
+            "desc":"Flags that an asset is offered for sale and sets address authorized to submit the transfer",
+            "args":[
+                {"name":"royalty_asset", "type":"asset"}, 
+                {"name":"royalty_asset_amount", "type":"uint64"}, 
+                {"name":"authorized_address", "type":"account"}
+            ],
+            "returns":{"type":"void"}
+        },
+        {
+            "name":"rescind",
+            "desc":"Rescinds offered asset",
+            "args":[
+                {"name":"royalty_asset", "type":"asset"}
+            ],
+            "returns":{"type":"void"}
+        },
+        {
+            "name":"move",
+            "desc":"Moves the asset passed from one account to another. This will wipe any offered state from the `from` account",
+            "args":[
+                {"name":"royalty_asset", "type":"asset"}, 
+                {"name":"royalty_asset_amount", "type":"uint64"}, 
+                {"name":"from", "type":"account"},
+                {"name":"to", "type":"account"}
+            ],
+            "returns":{"type":"void"}
+        },
+        {
+            "name":"set_policy",
+            "desc":"Sets the royalty policy for a given app id",
+            "args":[
+                {"name":"royalty_asset", "type":"asset"}, 
+                {"name":"royalty_receiver", "type":"address"},
+                {"name":"royalty_share", "type":"uint64"},
+                {"name":"allowed_asset_1", "type":"asset"},
+                {"name":"allowed_asset_2", "type":"asset"},
+                {"name":"allowed_asset_3", "type":"asset"},
+                {"name":"allowed_asset_4", "type":"asset"}
+            ],
+            "returns":{"type":"void"}
+        },
+        {
+            "name":"transfer",
+            "desc":"Transfers an Asset from one account to another and enforces royalty payments. If payment is in Algos, payment_asset should be 0",
+            "args":[
+                {"name":"royalty_asset", "type":"asset"}, 
+                {"name":"owner_acct", "type":"account"},
+                {"name":"buyer_acct", "type":"account"},
+                {"name":"royalty_receiver", "type":"account"},
+                {"name":"royalty_asset_amount", "type":"uint64"},
+                {"name":"payment", "type":"txn"},
+                {"name":"payment_asset", "type":"asset"}
+            ],
+            "returns":{"type":"void"}
+        }
+    ],
+    "locals": [ 
+        {
+            "name": "{uint64 asset id as big endian bytes}",
+            "desc": "The ID of the asset that has been offered as key, authorizing address as value",
+            "type": "(address,uint64)"
+        }  
+    ],
+    "globals": [ 
+        {
+            "name": "0000000000000000",
```

Comment:
Is it really this or the 0 byte to be consistent with the royalty policy for asset ID in key

Reply:
See also https://github.com/algorandfoundation/ARCs/pull/75#discussion_r839006561

Reply:
In the current implementation, its 8 0 bytes representing the uint64 zero value to be consistent. We could rename to just `default` or something?